### PR TITLE
NEWRELIC-5601 Example messaging

### PR DESCRIPTION
**Pull request by jordigh**
This is a toy messaging app, exercised via an express app and instrumented separately in its own `instrumentation.js` module. It shows how to use `instrumentMessages` from the main API and how to use `recordConsume`, `recordSubscribedConsumed` and `recordProduce` from the message shim API.

The README should include enough details to explain what everything is.

### CODE REVIEW

File: messaging-app/instrumentation.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+'use strict'
+
+const newrelic = require('newrelic')
+
+newrelic.instrumentMessages('./messages', (shim, messages, modname) => {
+  console.log(`[NEWRELIC] instrumenting ${modname}`)
+  const Client = messages.Client
+  shim.setLibrary('messages')
+
+  console.log(`[NEWRELIC] instrumenting method 'publish'`)
```
[the remark this answers is not in the document]
Comment (jordigh) [reply messaging-app/instrumentation.js:15]: I kind of deliberately wanted there to be stuff to the console. I feel that's more instructive and immediately obvious.

File: messaging-app/instrumentation.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ */
+
+'use strict'
+
+const newrelic = require('newrelic')
+
+newrelic.instrumentMessages('./messages', (shim, messages, modname) => {
+  console.log(`[NEWRELIC] instrumenting ${modname}`)
+  const Client = messages.Client
+  shim.setLibrary('messages')
```
[the remark this answers is not in the document]
Comment (jordigh) [reply messaging-app/instrumentation.js:13]: Sure. I mean the library *is* named `messages`, but I can change the name to make it clearer.

File: messaging-app/index.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      res.send([`Subscribed to queue ${queueName}`])
+    },
+    function consumeMessage(msg) {
+      if (messagesFromSub[queueName] === undefined) {
+        messagesFromSub[queueName] = []
+      }
+      messagesFromSub[queueName].push(msg)
+    }
+  )
+  res.end()
```
[the remark this answers is not in the document]
Comment (jordigh) [reply messaging-app/index.js:70]: I'll write something more about this in the README. It is a bit tricky.

File: messaging-app/instrumentation.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bizob2828) [reply messaging-app/instrumentation.js:15]: no worries.

File: messaging-app/instrumentation.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (bizob2828) [reply messaging-app/instrumentation.js:13]: i found it very confusing at first glance of transaction names. So was just thinking out loud about conveying a little more about what that part of the path segment means

File: messaging-app/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jordigh) [reply messaging-app/index.js:70]: By the way, this is what I was asking last week about when transactions should be created when handling a subscribed message. The only thing that made sense to me in the end is that handling of messages happened within a different transaction. If you're subscribed, the transaction that publishes messages is also the transaction that handles messages.

File: messaging-app/instrumentation.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jordigh) [reply messaging-app/instrumentation.js]: The library is now called Nifty Messages, hopefully it makes it clear that's what we're instrumenting and naming.

File: messaging-app/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jordigh) [reply messaging-app/index.js]: I've added a comment to the instrumentation that hopefully makes this clearer.

File: messaging-app/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+The application consists of the following files.
+
+* [`messages.js`](./messages.js): a toy messaging app that exposes three prototypical methods, `publish`, `subscribe`, and `getMessage`.
+* [`index.js`](./index.js): a simple Express app that makes use of the toy messaging app.
+* [`instrumentation.js`](./instrumentation.js): all of the New Relic instrumentation is in here. The `npm start` command loads this module first.
+* [`newrelic.js`](./newrelic.js): a basic, sample New Relic configuration.
+
+The instrumentation in `instrumentation.js` makes use of [`instrumentMessages`](https://docs.newrelic.com/docs/apm/agents/nodejs-agent/api-guides/nodejs-agent-api/#instrumentMessages) and three relevant functions of the [messaging shim API](https://newrelic.github.io/node-newrelic/docs/MessageShim.html):
+
+* [`recordProduce`](https://newrelic.github.io/node-newrelic/docs/MessageShim.html#recordProduce)
```
[the remark this answers is not in the document]
Comment (jordigh) [reply messaging-app/README.md:104]: Done.

**Pull request merged**